forest/sycamore/responses.py

```diff
@@ -171,3 +185,115 @@ def get_most_common_answer(col):
 
     detail = detail[detail_cols]
     return detail, out
+
+
+def responses_by_submission(agg_data: pd.DataFrame) -> dict:
```

Comment:
It's customary to start a function name with an action verb that describes what a function actually does.